compiler/hash-tir/src/terms.rs

```diff
@@ -85,28 +95,45 @@ pub enum Term {
 
     // Casting
     Cast(CastTerm),
-
-    // Types
     TypeOf(TypeOfTerm),
-    Ty(TyId),
 
     // References
     Ref(RefTerm),
     Deref(DerefTerm),
 
+    // -- Types --
+    /// Tuple type
+    TupleTy(TupleTy),
+
+    /// Function type
+    FnTy(FnTy),
+
+    /// Reference type
+    RefTy(RefTy),
+
+    /// A user-defined data type
+    DataTy(DataTy),
+
+    /// The universe type
+    Universe,
+
     /// Holes
     Hole(Hole),
 }
 
 tir_node_single_store!(Term);
 tir_node_sequence_store_indirect!(TermList[TermId]);
 
+pub type Ty = Term;
+pub type TyId = TermId;
+pub type TyListId = TermListId;
+
```

Comment:
Interesting hack, I was wondering how it was possible to have `Term` and `Ty` in the same match block.